### PR TITLE
Filters features

### PR DESCRIPTION
**Pull request by sescandell**
Some various fixes/improvements on filters:
 - [x] Let's filters take into account groups
 - [x] Manage filters fields order
 - [x] Fix groups management
 - [x] Manage filters on association fields
  - [x] ORM
  - [x] ODM
  - [x] Propel
 - [x] Making filters credentials different than list fields ones
 - [x] Let scopes able to filter on secured fields

Fix issues: #82 , #73 , #58 , #36

### CODE REVIEW

**Comment by loostro** [comment]
@sescandell Can we add one some feature to the list?

- [] add "show/hide" button
- [] make filters panel "hidden" by default (or let the "default" be configurable in generator.yml)

This would make "more space" for table columns. Current behaviour is fine if you have a few columns, but as soon as you display more than 8.. it's a nightmare.

**Comment by loostro** [comment]
@sescandell The changeset looks OK to me. Nice work!

**Comment by sescandell** [comment]
Ok for your first comment !

**Comment by loostro** [comment]
@sescandell i found a lightweight animation script, I've pushed a commit into your branch already

[here you can see demo](http://jsfiddle.net/v0Lasrxw/)

**Comment by sescandell** [comment]
I was more thinking about using AdminLTE integrated solution for that... But let's see ho we can integrate it.

**Comment by sescandell** [comment]
I should finish this PR this Week-end

**Comment by loostro** [comment]
@sescandell i was looking for adminlte solution, but im not sure the `sidebar` markup will allow any size elements

if you can make it useing adminlte i'm :+1: for that

**Comment by sescandell** [comment]
Hi @loostro ,

I didn't notice your JS file was about "slided" boxes.
I start to work on your proposition through commit #5b76066 but it requires some additional work on how to display such a box.
So I propose to move this work in another PR. I implemented this behavior for the "top" mode filters. We just need to handle the side one.

I would prefer to get some feedback for all other modifications, to merge this PR than focus on slided filters box (I was also thinking about re-using the adminlte config panel like here: https://almsaeedstudio.com/preview (see the parameter panel on right: sounds like a very good solution for filters, what do you think?)

@tobias-93 : don't hesitate to provide any feedback too.

On my side, I'm going to test all of this over the demo bundle (and rebase the PR... but don't know yet how...)

**Comment by sescandell** [comment]
Rebase done... (but all commits have been squashed to one... I've made a miss command... not sure this is really important).

**Comment by sescandell** [comment]
I have an issue on collections fields... I'm working on it

**Comment by loostro** [comment]
@sescandell OK I'm opening a new issue on this#115

File: Guesser/PropelORMFieldGuesser.php [hunk 1]
```diff
@@ -8,23 +8,34 @@
...
+        return $this->getTable($class);
     }
 
+    /**
+     * @param $class
+     * @return array
+     */
     public function getAllFields($class)
     {
         $return = array();
```
Comment (loostro) [inline Guesser/PropelORMFieldGuesser.php]: @sescandell does this return all fields (also relations?)

**Comment by loostro** [comment]
@sescandell please rebase the PR to include #116 fixes

File: Guesser/PropelORMFieldGuesser.php (hunk 1 above)
Comment (sescandell) [inline Guesser/PropelORMFieldGuesser.php]: According to this: http://propelorm.org/documentation/cookbook/runtime-introspection.html... I'm not anymore sure :D but I think.

Experienced propel user are welcome to express themselves here

**Comment by sescandell** [comment]
It seems there is no need to rebase, Merge can be processed

**Comment by loostro** [comment]
Fixed failure asserting that `'entity' => 'model'` equals `'entity' => 'entity'`.

**Comment by sescandell** [comment]
Can we merge that one now?

**Comment by loostro** [comment]
@sescandell merged, thanks

**Pull request merged**